buildlogger/client.go

```diff
@@ -0,0 +1,279 @@
+package buildlogger
+
+import (
+	"context"
+	"crypto/tls"
+	"time"
+
+	"github.com/evergreen-ci/aviation"
+	"github.com/evergreen-ci/timber/buildlogger/internal"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+	"github.com/mongodb/grip/send"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+type buildlogger struct {
+	ctx        context.Context
+	opts       *BuildloggerOptions
+	conn       *grpc.ClientConn
+	client     internal.BuildloggerClient
+	buffer     []*internal.LogLine
```

Comment:
should we avoid the slice of pointers here.

Reply:
No because the gRPC function takes in a slice of pointers and we are never iterating through this buffer